lib/api/completeMultipartUpload.js

```diff
@@ -499,31 +504,44 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                                 }
                                 return next(null, mpuBucket, keysToDelete,
                                     aggregateETag, extraPartLocations,
-                                    destinationBucket, generatedVersionId);
+                                    destinationBucket, generatedVersionId,
+                                    droppedMPUSize);
                             });
                     }
                     return next(null, mpuBucket, keysToDelete, aggregateETag,
                         extraPartLocations, destinationBucket,
-                        generatedVersionId);
+                        generatedVersionId, droppedMPUSize);
                 });
         },
         function deletePartsMetadata(mpuBucket, keysToDelete, aggregateETag,
-            extraPartLocations, destinationBucket, generatedVersionId, next) {
+            extraPartLocations, destinationBucket, generatedVersionId, droppedMPUSize, next) {
```

Comment:
"droppedMPUSize" can be computed around here (sum of every `extraPartLocation`'s size), as it is the place where we drop the extra parts : code will be must simpler & easier to follow, with no risk of mistake between all the steps...

this also makes it easier to handle partial delete object metadata errors : i.e. update quota with only the parts which were succesfully deleted...